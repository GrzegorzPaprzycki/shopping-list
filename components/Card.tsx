import { FC, ReactNode } from "react";

type IProps = {
  className?: string;
  children: ReactNode;
};

const Card: FC<IProps> = ({ className, children }) => {
  return (
    <div
      className={`${className} rounded-3xl px-10 py-4 drop-shadow-xl bg-white`}>
      {children}
    </div>
  );
};

export default Card;
