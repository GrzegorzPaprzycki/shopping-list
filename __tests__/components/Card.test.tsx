import { render, screen } from "@testing-library/react";
import Card from "../../components/Card";
import "@testing-library/jest-dom";

describe("Card", () => {
  it("renders without errors", () => {
    // given
    // when
    const { container } = render(<Card>Test content</Card>);

    // then
    const proof = screen.getByText(/Test content/i);
    expect(proof).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
