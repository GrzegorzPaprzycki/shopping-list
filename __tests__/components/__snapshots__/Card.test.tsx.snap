// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders a heading 1`] = `
<div>
  <div
    class="undefined px-10 py-4 drop-shadow-xl bg-white"
  >
    Test content
  </div>
</div>
`;

exports[`Card renders without errors 1`] = `
<div>
  <div
    class="undefined px-10 py-4 drop-shadow-xl bg-white"
  >
    Test content
  </div>
</div>
`;
